# Nivell 1 (màxim 5 punts)

## Pregunta 1 (1 punt)

Executa el següent codi en una cel·la:

```java
for(byte b = -125; b >= -130 && b <= 0; b--)
    println(b)
```

In [3]:
//Escriu i executa el codi aquí
for(byte b = -125; b >= -130 && b <= 0; b--)
    println(b)

-125
-126
-127
-128


**Respon**:

1. Per què no s'imprimeix el -129 i el -130?
2. Per què cal posar la condició b <= 0 ?

---
Resposta 1: Porque tiene el de desde -128 a 127

Escriu la teva resposta aquí i **raona la teva resposta**.

IMPORTANT: cal raonar la resposta per poder puntuar.

---
Resposta 2: Si no ponemos la condicion  se va quedar en bucle infinito

Escriu la teva resposta aquí i **raona la teva resposta**.

IMPORTANT: cal raonar la resposta per poder puntuar.

## Pregunta 2 (1 punt)

Executa el següent codi en una cel·la:

```java
String lang2 = "Java";
String lang1 = "Hello";
println(lang1 + lang2);
println(lang1 + lang2 == "HelloJava");
```

In [1]:
//Escriu i executa el codi aquí
String lang2 = "Java";
String lang1 = "Hello";
println(lang1 + lang2);
println(lang1 + lang2 == "HelloJava");

HelloJava
false


**Respon**: 

Per què la última línia de codi imprimeix false?

---
Resposta:Porque en la memoria lang2 es "Java" y lang1 es "Hello" " print(lang1 + lang2)" te une el string y muestra por la pantalla pero no guarda en la memoria, por eso lang1 + lang2 === "HelloJava" da false

Escriu la teva resposta aquí i **raona la teva resposta**.

IMPORTANT: cal raonar la resposta per poder puntuar.

## Exercici 1 (1,5 punts)

Donada una cadena de text, a continuació teniu un snippet que canvia el primer caràcter pel segon:

In [14]:
String word = "java";

char[] letters = word.toCharArray();
char letter1 = letters[0];
char letter2 = letters[1];
letters[0] = letter2;
letters[1] = letter1;
print(new String(letters));


ajva

Aprofitant el codi anterior, feu un snippet que trobi totes les combinacions possibles que s'obtenen en intercanviar la primera lletra per cadascuna de les lletres restants.

Per exemple, per a "java" obtindrem:

```
ajva vaja aavj
```

El vostre codi ha de ser vàlid per a paraules de qualsevol mida.

In [15]:
//El vostre codi a continuació:
String word = "java";

for (int i = 1; i < word.length(); i++) { //un bucle k empieza de index 1 k va hasta tamaño de word
    char[] letters = word.toCharArray(); //convertir word en array de chars
    char firstLetter = letters[0]; //valor pordefecto 
    char currentLetter = letters[i];
    letters[0] = currentLetter;//swap
    letters[i] = firstLetter;//swap
    System.out.print(new String(letters) + " ");//mostrar por la pantalla
}


ajva vaja aavj 

## Exercici 2 (1,5 punts)

In [2]:
// Codi per a obtenir un array d'enters aleatoris

int getRandomNumber(int min, int max) {
    int upper = max * 10;
    Random random = new Random();
    return random.nextInt(upper - min) + min;
}

int[] createRandomIntArray(int max){
    int[] numbers = new int[max];
    for(int i = 0; i < max; i++){
        numbers[i] = getRandomNumber(0,max);
    }
    return numbers;
}

int[] numbers = createRandomIntArray(100);

print(Arrays.toString(numbers))

[672, 322, 882, 773, 344, 638, 556, 371, 828, 319, 815, 633, 439, 884, 642, 566, 849, 929, 692, 350, 623, 293, 39, 818, 719, 226, 633, 458, 408, 287, 906, 2, 378, 43, 34, 979, 22, 598, 342, 740, 829, 760, 919, 246, 157, 85, 506, 306, 415, 644, 230, 39, 983, 277, 894, 981, 951, 784, 612, 57, 755, 99, 239, 13, 618, 622, 794, 352, 474, 434, 837, 870, 950, 843, 897, 285, 595, 711, 74, 240, 329, 617, 515, 448, 276, 76, 696, 585, 999, 386, 903, 504, 186, 60, 13, 87, 957, 519, 843, 523]

***

Feu un snippet que, donat un array de números aleatoris com el generat amb el codi anterior, calculi **la mitjana** (que pot ser decimal) que s'obté de tots els números **parells** que pertanyen a l'array.

 

In [15]:
int[] numbers = createRandomIntArray(100);

//TODO (solució)
  // getting array length
        int length = numbers.length;
 
        // valor por fecto de sum
    
 int sum = 0;
int count = 0;

for(int i = 0; i < numbers.length; i++){
    if(numbers[i] % 2 == 0){
        sum += numbers[i]; //sumar los numeros parells
        count++;
    }
}
// comprobar si es parell
if(count > 0){
    double average = (double) sum / count;
    System.out.println("Mitjana dels nombres parells: " + average);
} else {
    System.out.println("No hi ha nombres parells a l'array.");
}





//TODO (jocs de proves)


Mitjana dels nombres parells: 549.9069767441861
